### DISCUSSION TITLE
DMG Levels

### DISCUSSION THREAD

**Discussion by akrabu** [aonez/Keka#1226] [Q&A]
Hello!  I was wondering what algorithm / compression levels the DMG slider corresponded to.  I was hoping to make several LZFSE DMG's but am not sure what corresponds to what.  Also if LZFSE isn't an option...  it would be cool if LZFSE (ULFO) and LZMA (ULMO) were added below "Use Bzip2 compression" (or perhaps have a selector for the algorithm of choice).

**Comment by aonez** [comment]
Currently those are the levels of compression, using `zlib` (GZIP by default, you can use BZIP2 if checked):

- Store: no compression, read-only image
- Fastest: 1
- Fast: 4
- Normal: 6
- Slow: 8
- Slowest: 9

I'll take a look and probably add support to those formats, since I'm seeing BZIP2 is being deprecated.

**Comment by aonez** [comment]
Sadly I'm not seeing any level customisation using these formats.

**Comment by akrabu** [comment]
Ahh no, LZFSE and LZVN don't have any levels.  Maybe make it so you could check LZFSE and it grays out the slider when you do??

**Comment by aonez** [comment]
It will be like this

<img width="385" alt="Screenshot 2023-04-20 at 13 06 23" src="https://user-images.githubusercontent.com/480163/233348000-77286644-41c4-4754-80f1-0a8a7f3d2b15.png">

**Reply by akrabu** [reply]
This is exactly what I envisioned.  Looks great!  I'll start testing it out now.  :)

**Reply by akrabu** [reply]
Out of curiosity, what does the slider actually do when LZFSE is selected?  I think UDZO (zlib) is the only ones with compression levels...  I don't even see UDBZ (Bzip2) as having levels in the man page?  🤔

Should the slider just be greyed out for formats that don't support levels?

**Comment by aonez** [comment]
In fact here it is: [Keka-v1.2.62.r5161](https://github.com/aonez/Keka/releases/download/dev-test-builds/Keka-v1.2.62.r5161.7z)

**Comment by aonez** [comment]
It should be greyed out on all formats but ZLIB and BZIP2, it isn’t for you?

The level setting only works with ZLIB and BZIP2 (this one I found out
there, bzip2-level). They both are 0-9 levels. But BZIP2 0 level is kind of
a default, not “no compression”.

**Reply by akrabu** [reply]
Oh that's weird.  It wasn't, but now it is.  See here https://v.usetapes.com/ohJu5WtxwO

I'll play with it and see if I can't get it to bug out again.

**Reply by akrabu** [reply]
Oh.  If you close it with LZFSE selected, and reopen it - it's slidable.

**Reply by aonez** [reply]
Indeed! Fixed that :)
